packages/google-compute-engine-oslogin/utils/oslogin_utils.cc

```diff
@@ -568,6 +655,129 @@ bool ParseJsonToChallenges(const string& json,
   return true;
 }
 
+bool FindGroup(struct group* result, BufferManager* buf, int* errnop) {
+  if (result->gr_name == NULL && result->gr_gid == 0) {
+    return false;
+  }
+  std::stringstream url;
+  std::vector<Group> groups;
+
+  string response;
+  long http_code;
+  string pageToken = "";
+
+  do {
+    url.str("");
+    url << kMetadataServerUrl << "groups";
+    if (pageToken != "")
+      url << "?pageToken=" << pageToken;
+
+    response.clear();
+    http_code = 0;
+    if (!HttpGet(url.str(), &response, &http_code) || http_code != 200 ||
+        response.empty()) {
+      *errnop = EAGAIN;
+      return false;
+    }
+
+    if (!ParseJsonToKey(response, "pageToken", &pageToken)) {
```

Comment:
In the response, the key is nextPageToken for both the pagination schemes. Can you have all the strings as constants so they are easy to read?